### PR TITLE
Osama/add wordpress integration

### PR DESCRIPTION
**Pull request by asamolion**

### CODE REVIEW

File: repo.sh [hunk 1]
```diff
@@ -41,7 +41,16 @@ private_repos=(
     "https://github.com/edx/edx-themes.git"
 )
 
+wp_plugin_repos=(
+    "git@github.com:edly-io/edly-wp-plugin.git"
+)
+
+wp_theme_repos=(
+    "git@github.com:edly-io/edly-wp-theme.git"
+)
```
Comment (Ayub-Khan) [inline repo.sh:50]: Kindly add those plugins too.
1. Elementor
2. Classic Editor
3. Open edX LMS and WordPress integrator
4. Advanced Custom Fields Pro

File: docker-compose-host.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ services:
       - credentials_node_modules:/edx/app/credentials/credentials/node_modules
       - ${DEVSTACK_WORKSPACE}/src:/edx/src:cached
   discovery:
-      volumes:
+    volumes:
```
Comment (Ayub-Khan) [inline docker-compose-host.yml:10]: Need to make this repo private.

File: wordpress/wp-config.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+ *
+ * For information on other constants that can be used for debugging,
+ * visit the Codex.
+ *
+ * @link https://codex.wordpress.org/Debugging_in_WordPress
+ */
+define( 'WP_DEBUG', true );
+
+/* That's all, stop editing! Happy publishing. */
+
```
Comment (Ayub-Khan) [inline wordpress/wp-config.php]: edunext plugin configurations.

**Review by Ayub-Khan — CHANGES_REQUESTED** [review]
Some permission issues you are setting www-data as the owner of the whole. which will be a problem as the people will be editing the code out side the docker which will be owned by the www-data user. while if you change that your wordpress wont work. 
i had to change own out side the repo to give both myself and www-data access.

**Comment by Ayub-Khan** [comment]
set the edly-wp-theme to default theme.

File: repo.sh (hunk 1 above)
Comment (asamolion) [reply repo.sh:50]: Why do we need the classic editor?

File: docker-compose-host.yml (hunk 2 above)
Comment (asamolion) [reply docker-compose-host.yml:10]: We can't make this private since this is forked from an open source repo.

File: repo.sh (hunk 1 above)
Comment (asamolion) [reply repo.sh:50]: Confirmed from Amjad bhai. Will be adding that as well.

Comment (asamolion) [reply repo.sh:50]: Done.

File: docker-compose-host.yml (hunk 2 above)
Comment (asamolion) [reply docker-compose-host.yml:10]: Also, I've made sure that our custom code isn't added to the docker container and that no one can clone it unless they are given explicit access on github. So no worries there.

**Comment by asamolion** [comment]
> set the edly-wp-theme to default theme.

That is not easily doable with the current config. I've added documentation stating that as a compulsory manual step after first setup.

**Comment by asamolion** [comment]
> Some permission issues you are setting www-data as the owner of the whole. which will be a problem as the people will be editing the code out side the docker which will be owned by the www-data user. while if you change that your wordpress wont work.
> i had to change own out side the repo to give both myself and www-data access.

The www-data permissions are by design. They are needed so that WordPress is able to function normally.

File: docker-compose-host.yml (hunk 2 above)
Comment (Ayub-Khan) [reply docker-compose-host.yml:10]: :+1:

**Review by Ayub-Khan — APPROVED** [review]
Looks good, Kindly fix conflicts and squash commits into one with a detailed message of Changes ie.

Edly Wordpress Integration:
-added clone feature for edly wp repos
-added wordpress clone support
...
...

File: wordpress/wp-config.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+/** Database Charset to use in creating database tables. */
+define( 'DB_CHARSET', 'utf8mb4' );
+
+/** The Database Collate type. Don't change this if in doubt. */
+define( 'DB_COLLATE', '' );
+
+
+/* That's all, stop editing! Happy blogging. */
+// Edly specific settings
+define('DISCOVERY_API_URL', 'edx.devstack.discovery:18381/api/v1');
```
Comment (Ayub-Khan) [inline wordpress/wp-config.php]: set only Discovery Base url remove api/v1

Comment (asamolion) [reply wordpress/wp-config.php]: Has the variable name changed?

**Comment by NIXKnight** [comment]
Why replace double quotes with single quotes on existing configuration?

**Comment by NIXKnight** [comment]
Isn't it better to use the `wp-config.php` file that is used in production?

**Comment by asamolion** [comment]
> Why replace double quotes with single quotes on existing configuration?

My editor did that automatically. If it is an issue I can revert.

**Comment by asamolion** [comment]
> Isn't it better to use the `wp-config.php` file that is used in production?

What do you mean by this?

**Comment by NIXKnight** [comment]
> > Why replace double quotes with single quotes on existing configuration?
> 
> My editor did that automatically. If it is an issue I can revert.

If its working fine its not an issue. Fix your editor though.

**Comment by NIXKnight** [comment]
> > Isn't it better to use the `wp-config.php` file that is used in production?
> 
> What do you mean by this?

I mean this:

```
<?php
// ** Site URL ** //
define( 'WP_HOME', getenv('WP_HOME') );
define( 'WP_SITEURL', getenv('WP_SITEURL') );

// ** MySQL Settings ** //
/** The name of the database for WordPress */
define( 'DB_NAME', getenv('DB_NAME') );
/** MySQL database username */
define( 'DB_USER', getenv('DB_USER') );
/** MySQL database password */
define( 'DB_PASSWORD', getenv('DB_PASSWORD') );
/** MySQL hostname */
define( 'DB_HOST', getenv('DB_HOST') );
/** Database Charset to use in creating database tables. */
define( 'DB_CHARSET', getenv('DB_CHARSET') );
/** The Database Collate type. Don't change this if in doubt. */
define( 'DB_COLLATE', '' );

// ** Authentication Unique Keys and Salts ** //
define( 'AUTH_KEY', getenv('AUTH_KEY') );
define( 'SECURE_AUTH_KEY', getenv('SECURE_AUTH_KEY') );
define( 'LOGGED_IN_KEY', getenv('LOGGED_IN_KEY') );
define( 'NONCE_KEY', getenv('NONCE_KEY') );
define( 'AUTH_SALT', getenv('AUTH_SALT') );
define( 'SECURE_AUTH_SALT', getenv('SECURE_AUTH_SALT') );
define( 'LOGGED_IN_SALT', getenv('LOGGED_IN_SALT') );
define( 'NONCE_SALT', getenv('NONCE_SALT') );

// ** WordPress Database Table Prefix ** //
$table_prefix = getenv('TABLE_PREFIX');

// ** WordPress debugging mode ** //
define( 'WP_DEBUG', false );

// ** EDLY Open edX Integration ** //
define( 'DISCOVERY_API_URL', getenv('DISCOVERY_API_URL') );
define( 'LMS_BASE_URL', getenv('LMS_BASE_URL') );
define( 'EOX_CLIENT_ID', getenv('EOX_CLIENT_ID') );
define( 'EOX_CLIENT_SECRET', getenv('EOX_CLIENT_SECRET') ); 
define( 'IS_LOGGED_IN_COOKIE', getenv('IS_LOGGED_IN_COOKIE') );
define( 'USER_INFO_COOKIE', getenv('USER_INFO_COOKIE') );

// ** S3 Bucket Configration //
define ( 'S3_UPLOADS_BUCKET', getenv('S3_UPLOADS_BUCKET') );

// ** Absolute Path to the WordPress Directory ** //
if ( ! defined( 'ABSPATH' ) ) {
  define( 'ABSPATH', dirname( __FILE__ ) . '/' );
}

// ** Sets up WordPress vars and included files. ** //
require_once( ABSPATH . 'wp-settings.php' );
```

**Comment by asamolion** [comment]
> > > Why replace double quotes with single quotes on existing configuration?
> > 
> > 
> > My editor did that automatically. If it is an issue I can revert.
> 
> If its working fine its not an issue. Fix your editor though.

Will do.

**Comment by asamolion** [comment]
> > > Isn't it better to use the `wp-config.php` file that is used in production?
> > 
> > 
> > What do you mean by this?
> 
> I mean this:
> 
> ```
> <?php
> // ** Site URL ** //
> define( 'WP_HOME', getenv('WP_HOME') );
> define( 'WP_SITEURL', getenv('WP_SITEURL') );
> 
> // ** MySQL Settings ** //
> /** The name of the database for WordPress */
> define( 'DB_NAME', getenv('DB_NAME') );
> /** MySQL database username */
> define( 'DB_USER', getenv('DB_USER') );
> /** MySQL database password */
> define( 'DB_PASSWORD', getenv('DB_PASSWORD') );
> /** MySQL hostname */
> define( 'DB_HOST', getenv('DB_HOST') );
> /** Database Charset to use in creating database tables. */
> define( 'DB_CHARSET', getenv('DB_CHARSET') );
> /** The Database Collate type. Don't change this if in doubt. */
> define( 'DB_COLLATE', '' );
> 
> // ** Authentication Unique Keys and Salts ** //
> define( 'AUTH_KEY', getenv('AUTH_KEY') );
> define( 'SECURE_AUTH_KEY', getenv('SECURE_AUTH_KEY') );
> define( 'LOGGED_IN_KEY', getenv('LOGGED_IN_KEY') );
> define( 'NONCE_KEY', getenv('NONCE_KEY') );
> define( 'AUTH_SALT', getenv('AUTH_SALT') );
> define( 'SECURE_AUTH_SALT', getenv('SECURE_AUTH_SALT') );
> define( 'LOGGED_IN_SALT', getenv('LOGGED_IN_SALT') );
> define( 'NONCE_SALT', getenv('NONCE_SALT') );
> 
> // ** WordPress Database Table Prefix ** //
> $table_prefix = getenv('TABLE_PREFIX');
> 
> // ** WordPress debugging mode ** //
> define( 'WP_DEBUG', false );
> 
> // ** EDLY Open edX Integration ** //
> define( 'DISCOVERY_API_URL', getenv('DISCOVERY_API_URL') );
> define( 'LMS_BASE_URL', getenv('LMS_BASE_URL') );
> define( 'EOX_CLIENT_ID', getenv('EOX_CLIENT_ID') );
> define( 'EOX_CLIENT_SECRET', getenv('EOX_CLIENT_SECRET') ); 
> define( 'IS_LOGGED_IN_COOKIE', getenv('IS_LOGGED_IN_COOKIE') );
> define( 'USER_INFO_COOKIE', getenv('USER_INFO_COOKIE') );
> 
> // ** S3 Bucket Configration //
> define ( 'S3_UPLOADS_BUCKET', getenv('S3_UPLOADS_BUCKET') );
> 
> // ** Absolute Path to the WordPress Directory ** //
> if ( ! defined( 'ABSPATH' ) ) {
>   define( 'ABSPATH', dirname( __FILE__ ) . '/' );
> }
> 
> // ** Sets up WordPress vars and included files. ** //
> require_once( ABSPATH . 'wp-settings.php' );
> ```

No, because we don't need these many config options in devstack. Also, it increases the changes of leaking keys while updating devstack. I believe a dummy wp-config is the best option for local development.

**Comment by NIXKnight** [comment]
> > > > Isn't it better to use the `wp-config.php` file that is used in production?
> > > 
> > > 
> > > What do you mean by this?
> > 
> > 
> > I mean this:
> > ```
> > <?php
> > // ** Site URL ** //
> > define( 'WP_HOME', getenv('WP_HOME') );
> > define( 'WP_SITEURL', getenv('WP_SITEURL') );
> > 
> > // ** MySQL Settings ** //
> > /** The name of the database for WordPress */
> > define( 'DB_NAME', getenv('DB_NAME') );
> > /** MySQL database username */
> > define( 'DB_USER', getenv('DB_USER') );
> > /** MySQL database password */
> > define( 'DB_PASSWORD', getenv('DB_PASSWORD') );
> > /** MySQL hostname */
> > define( 'DB_HOST', getenv('DB_HOST') );
> > /** Database Charset to use in creating database tables. */
> > define( 'DB_CHARSET', getenv('DB_CHARSET') );
> > /** The Database Collate type. Don't change this if in doubt. */
> > define( 'DB_COLLATE', '' );
> > 
> > // ** Authentication Unique Keys and Salts ** //
> > define( 'AUTH_KEY', getenv('AUTH_KEY') );
> > define( 'SECURE_AUTH_KEY', getenv('SECURE_AUTH_KEY') );
> > define( 'LOGGED_IN_KEY', getenv('LOGGED_IN_KEY') );
> > define( 'NONCE_KEY', getenv('NONCE_KEY') );
> > define( 'AUTH_SALT', getenv('AUTH_SALT') );
> > define( 'SECURE_AUTH_SALT', getenv('SECURE_AUTH_SALT') );
> > define( 'LOGGED_IN_SALT', getenv('LOGGED_IN_SALT') );
> > define( 'NONCE_SALT', getenv('NONCE_SALT') );
> > 
> > // ** WordPress Database Table Prefix ** //
> > $table_prefix = getenv('TABLE_PREFIX');
> > 
> > // ** WordPress debugging mode ** //
> > define( 'WP_DEBUG', false );
> > 
> > // ** EDLY Open edX Integration ** //
> > define( 'DISCOVERY_API_URL', getenv('DISCOVERY_API_URL') );
> > define( 'LMS_BASE_URL', getenv('LMS_BASE_URL') );
> > define( 'EOX_CLIENT_ID', getenv('EOX_CLIENT_ID') );
> > define( 'EOX_CLIENT_SECRET', getenv('EOX_CLIENT_SECRET') ); 
> > define( 'IS_LOGGED_IN_COOKIE', getenv('IS_LOGGED_IN_COOKIE') );
> > define( 'USER_INFO_COOKIE', getenv('USER_INFO_COOKIE') );
> > 
> > // ** S3 Bucket Configration //
> > define ( 'S3_UPLOADS_BUCKET', getenv('S3_UPLOADS_BUCKET') );
> > 
> > // ** Absolute Path to the WordPress Directory ** //
> > if ( ! defined( 'ABSPATH' ) ) {
> >   define( 'ABSPATH', dirname( __FILE__ ) . '/' );
> > }
> > 
> > // ** Sets up WordPress vars and included files. ** //
> > require_once( ABSPATH . 'wp-settings.php' );
> > ```
> 
> No, because we don't need these many config options in devstack. Also, it increases the changes of leaking keys while updating devstack. I believe a dummy wp-config is the best option for local development.

ok proceed then.

**Pull request merged**